subsys/bluetooth/services/fast_pair/fp_bond_manager.c

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <errno.h>
+
+#include <bluetooth/services/fast_pair/fast_pair.h>
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_DECLARE(fast_pair, CONFIG_BT_FAST_PAIR_LOG_LEVEL);
+
+#include "fp_activation.h"
+#include "fp_storage_ak.h"
+
+struct bond_find_ctx {
+	bool bond_found;
+	const bt_addr_le_t *bond_addr;
+};
+
+static void identity_resolved(struct bt_conn *conn, const bt_addr_le_t *rpa,
+			      const bt_addr_le_t *identity)
+{
+	if (!bt_fast_pair_is_ready()) {
+		return;
+	}
+
+	LOG_DBG("Identity resolved");
+
+	fp_storage_ak_bond_update(conn, identity);
+}
+
+static void disconnected(struct bt_conn *conn, uint8_t reason)
+{
+	if (!bt_fast_pair_is_ready()) {
+		return;
+	}
+
+	fp_storage_ak_bond_stopped(conn);
+}
+
+BT_CONN_CB_DEFINE(conn_callbacks) = {
+	.disconnected = disconnected,
+	.identity_resolved = identity_resolved,
+};
+
+static void bond_address_cmp(const struct bt_bond_info *info, void *user_data)
+{
+	struct bond_find_ctx *ctx = user_data;
+
+	if (ctx->bond_found) {
+		return;
+	}
+
+	if (bt_addr_le_eq(ctx->bond_addr, &info->addr)) {
+		ctx->bond_found = true;
+	}
+}
+
+static int bond_identity_find(uint8_t *identity, const bt_addr_le_t *addr)
+{
+	struct bond_find_ctx ctx = {
+		.bond_found = false,
+		.bond_addr = addr,
+	};
+
+	for (uint8_t id = 0; id < CONFIG_BT_ID_MAX; id++) {
+		bt_foreach_bond(id, bond_address_cmp, &ctx);
+		if (ctx.bond_found) {
+			if (identity) {
+				*identity = id;
+			}
+			return 0;
+		}
+	}
+
+	return -ESRCH;
+}
+
+static int bond_remove(const bt_addr_le_t *addr)
+{
+	uint8_t identity;
+	int err;
+
+	err = bond_identity_find(&identity, addr);
+	if (err) {
+		return err;
+	}
+
+	return bt_unpair(identity, addr);
+}
+
+static int bond_find(const bt_addr_le_t *addr)
+{
+	return bond_identity_find(NULL, addr);
+}
+
+static const struct fp_storage_ak_bond_bt_request_cb bt_request_cb = {
+	.bond_remove = bond_remove,
+	.bond_find = bond_find,
+};
+
+static int bond_manager_init(void)
+{
+	fp_storage_ak_bond_bt_request_cb_register(&bt_request_cb);
```

Comment:
Since `bond_find` callback is called only during intialization, I would suggest to make it an API function. - e.g. `fp_storate_ak_bond_load_ble_bond(address)`.

Reply:
Then we would need to invert the initialization priorities and delay calling `bond_remove` callbacks until the `fp_storage_ak_bond_bt_request_cb_register` is called

Reply:
Discussed offline with @alstrzebonski and @kapi-no. We will stick with the existing approach.